cpp/mcap/include/mcap/writer.inl

```diff
@@ -743,10 +750,17 @@ void McapWriter::writeChunk(IWritable& output, IChunkWriter& chunkData) {
     chunkIndexRecord.uncompressedSize = uncompressedSize;
   } else if (!options_.noMessageIndex) {
     // Write the message index records
-    for (const auto& [channelId, messageIndex] : currentMessageIndex_) {
-      write(output, messageIndex);
+    for (auto& [channelId, messageIndex] : currentMessageIndex_) {
```

Comment:
Seems like maybe this whole block should be deduped with the above – can we move the writing of message index outside the `if (!options_.noChunkIndex)` so it doesn't have to happen in two places?